runners/direct-java/src/main/java/org/apache/beam/runners/direct/ExecutorServiceParallelExecutor.java

```diff
@@ -370,14 +365,18 @@ public void run() {
         for (ExecutorUpdate update : updates) {
           LOG.debug("Executor Update: {}", update);
           if (update.getBundle().isPresent()) {
-            scheduleConsumers(update);
+            if (ExecutorState.ACTIVE == startingState
```

Comment:
Can you add comments on the logic above (the transitions) and here? Specifically:

Why does a draining pipeline with no work outstanding need to schedule consumers? Shouldn't that be drained? etc.